SmartDeviceLink-iOS/SmartDeviceLink/SDLHexUtility.m

```diff
@@ -7,16 +7,32 @@
 
 @implementation SDLHexUtility
 
-+ (NSString *)getHexString:(UInt8 *)bytes length:(int)length {
-    NSMutableString *ret = [NSMutableString stringWithCapacity:(length * 2)];
-    for (int i = 0; i < length; i++) {
-        [ret appendFormat:@"%02X", ((Byte *)bytes)[i]];
+static inline char itoh(int i) {
+    if (i > 9) return 'A' + (i - 10);
+    return '0' + i;
+}
+
+NSString* getHexString(NSData *data) {
+    NSUInteger i, len;
+    unsigned char *buf, *bytes;
+    
+    len = data.length;
+    bytes = (unsigned char*)data.bytes;
+    buf = malloc(len*2);
+    
+    for (i=0; i<len; i++) {
+        buf[i*2] = itoh((bytes[i] >> 4) & 0xF);
+        buf[i*2+1] = itoh(bytes[i] & 0xF);
     }
-    return ret;
+    
+    return [[NSString alloc] initWithBytesNoCopy:buf
+                                          length:len*2
+                                        encoding:NSASCIIStringEncoding
+                                    freeWhenDone:YES];
```

Comment:
If this fails, then the buffer isn't freed. You'll need to check that the return of the `init` method is not nil. If it is, call `free(buf)` and return nil.

See the "special considerations" on the apple documentation for the method.